plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -15,14 +15,26 @@ import (
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
+type helper struct {
+	sysctl   sysctlF
+	zpool    zpoolF
+	zdataset zdatasetF
+	uname    unameF
+}
+
+type sysctlF func(metric string) ([]string, error)
+type zpoolF func() ([]string, error)
+type zdatasetF func(properties []string) ([]string, error)
+type unameF func() (string, error)
```

Comment:
Do they really need to be variable functions? Can't they be used directly?

```diff
-	lines, err := z.zdataset(properties)
+	lines, err := zdataset(properties)
```

Reply:
Have you tried running the tests after this change? 🤔

Reply:
Okay, thanks for just nicely saying that it's needed to mock them for tests..